xarray/backends/h5netcdf_.py

```diff
@@ -121,6 +121,27 @@ def open(
     ):
         import h5netcdf
 
+        if isinstance(filename, bytes):
+            raise ValueError(
+                "can't open netCDF4/HDF5 as bytes "
+                "try passing a path or file-like object"
+            )
+        elif hasattr(filename, "tell"):
+            if filename.tell() != 0:
+                raise ValueError(
+                    "file-like object read/write pointer not at zero "
+                    "please close and reopen, or use a context "
+                    "manager"
+                )
```

Comment:
I think we could use a test for this case. Something like:

```python
with open(tmp_file, "rb") as f:
    f.seek(8)
    with raises_regex(ValueError, "file-like object read/write pointer not at zero"):
       open_dataset(f, engine="h5netcdf")

Reply:
The code path with an explicit engine is tested indirectly in `test_open_twice`. We could add the test with `engine=None` to exercise the paht in `api.py`, I'll add it.

Reply:
Mm, several lines are not covered, I'm adding test.

Reply:
Coverage is almost complete (only trivial things are missing IMO). Feel free to resolve this comment (or add to it!).

Reply:
Thanks for looking into it! All good here.